test/support/route.go

```diff
@@ -36,3 +40,45 @@ func GetRoute(t Test, namespace, name string) *routev1.Route {
 	t.T().Helper()
 	return Route(t, namespace, name)(t)
 }
+
+func ExposeService(t Test, namespace string, service string, route string, port string, expetedCode int) *routev1.Route {
+	r := &routev1.Route{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: routev1.SchemeGroupVersion.String(),
+			Kind:       "Route",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      route,
+			Namespace: namespace,
+		},
+		Spec: routev1.RouteSpec{
+			To: routev1.RouteTargetReference{
+				Name: service,
+			},
+			Port: &routev1.RoutePort{
+				TargetPort: intstr.FromString(port),
+			},
+		},
+	}
+
+	_, err := t.Client().Route().RouteV1().Routes(r.Namespace).Create(t.Ctx(), r, metav1.CreateOptions{})
+	t.Expect(err).NotTo(gomega.HaveOccurred())
+	t.T().Logf("Created Route %s/%s successfully", r.Namespace, r.Name)
+
+	t.T().Logf("Waiting for Route %s/%s to be available", r.Namespace, r.Name)
+	t.Eventually(Route(t, r.Namespace, r.Name), TestTimeoutLong).
+		Should(gomega.WithTransform(ConditionStatus(routev1.RouteAdmitted), gomega.Equal(corev1.ConditionTrue)))
+
+	// Retrieve hostname
+	r, err = t.Client().Route().RouteV1().Routes(r.Namespace).Get(t.Ctx(), r.Name, metav1.GetOptions{})
+	t.Expect(err).NotTo(gomega.HaveOccurred())
+	hostname := r.Status.Ingress[0].Host
+
+	// Wait for expected HTTP code
+	t.Eventually(func() int {
+		resp, _ := http.Get("http://" + hostname)
+		return resp.StatusCode
+	}, TestTimeoutLong).Should(gomega.Equal(expetedCode))
```

Comment:
What about waiting until Route stops returning HTTP 503 and a text in response body referring to the Pods being down?
That would allow dropping expected code parameter.